packages/compile-solidity/compilerSupplier/loadingStrategies/VersionRange.js

```diff
@@ -1,20 +1,68 @@
 const debug = require("debug")("compile:compilerSupplier");
 const requireFromString = require("require-from-string");
-const fs = require("fs");
 const originalRequire = require("original-require");
 const axios = require("axios").default;
 const semver = require("semver");
 const solcWrap = require("solc/wrapper");
-const LoadingStrategy = require("./LoadingStrategy");
+const Cache = require("../Cache");
+const observeListeners = require("../observeListeners");
+const { NoVersionError, NoRequestError } = require("../errors");
+
+class VersionRange {
+  constructor(options) {
+    const defaultConfig = {
+      compilerRoots: [
+        "https://relay.trufflesuite.com/solc/bin/",
+        "https://solc-bin.ethereum.org/bin/",
+        "https://ethereum.github.io/solc-bin/bin/"
```

Comment:
Probably separate PR... I'll make an issue for it

Reply:
Issue opened [here](https://github.com/trufflesuite/truffle/issues/4252)

Reply:
https://ethereum.github.io/solc-bin/bin/ is deprecated but it's just not receiving new binaries. Old ones are still there. Keeping it here should be fine for the time being and could actually provide an additional fallback in case the other options go down. It would only be a problem if we needed to rebuild one of the binaries and replace it but these cases are rare (currently there's just one that might end up being rebuilt eventually: https://github.com/ethereum/solidity/issues/11671).

Also, you might want to add https://solc-bin.ethereum.org/wasm/ (or rather https://binaries.soliditylang.org/wasm/ but it's just a cosmetic difference as they both point at the same thing) to the list above `bin/`. It has wasm rather than asm.js builds and uses the same naming convention so it can be used as the primary source with `bin/` being a fallback. Wasm builds offer better performance and don't have problems like https://github.com/ethereum/solidity/issues/11671. In 0.6.2 we switched to doing wasm builds and putting them in `bin/` but older binaries were not replaced. Instead wasm rebuilds of older versions are in `wasm/`.

To make this more apparent there are new dirs that contain the same binaries but correctly separated (`emscripten-asmjs/` and `emscripten-wasm32/`). I'd recommend using them if possible, though unfortunately they are not drop-in replacements since the platform name is now a part of the naming scheme.

Note also that each directory, including `wasm/`, has a separate `list.json` and wasm binaries obviously have different hashes than the asm.js ones so maybe it does not fit into this system after all. At least if you do not download `list.json` from the same entry you use to download binaries.

Reply:
Thanks @cameel ! 

cc: @gnidan